App.js

```diff
@@ -13,68 +13,99 @@ import styles from "./styles";
 
 const rnSentianceEmitter = new NativeEventEmitter(RNSentiance);
 
+const userLinkFlag = 'SDK_USER_LINKED';
+
 export default class App extends Component {
   state = {
     userId: "...",
     sdkVersion: "...",
     userActivityText: "...",
+    userLinkInstallId: "...",
     data: [],
-    subscriptionsAdded: false
   };
 
   async componentDidMount() {
     try {
-      await this.setupSdk();
-      this.subscribeAfterSDKSetup();
+      this.subscribeSDKEvents();
+      await this.initSDK();
       await this.requestPermissionAndroid();
     } catch (e) {
       console.error(e.code, e);
     }
   }
 
   componentWillUnmount() {
-    if (this.state.subscriptionsAdded) {
-      this.unSubscribe();
-    }
+    this.unSubscribe();
   }
 
   unSubscribe() {
     this.sdkStatusSubscription.remove();
     this.sdkUserActivityUpdateSubscription.remove();
+    this.userLinkListener.remove();
+  }
+
+  /**
+   * The user-linking logics are required to be implemented in your backend.
+   * 1. Sending the installId to your backend.
+   * 2. Calling the Sentiance API with the installId and the externalId in your backend.
+   * 3. Responding the result.
+   * If the backend responds successful, call RNSentiance.userLinkCallback(true);. Otherwise, RNSentiance.userLinkCallback(false);
+   * SDK will not be initialized if linking is failed.
+   */
+  async linkMetaUser(installId) {
+    throw new Error('Please implement the user linking logics here then remove this line.');
```

Comment:
I would suggest this message instead: "You must implement your own user linking logic inside linkMetaUser(installId)".

Actually, it would also be better if we name the method linkUser instead of linkMetaUser, since the latter is a deprecated terminology.

Reply:
Updated.